azurerm/resource_arm_virtual_wan_test.go

```diff
@@ -0,0 +1,216 @@
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Hello", "There"),
+					resource.TestCheckResourceAttr(resourceName, "tags.World", "Example"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testCheckAzureRMVirtualWanDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*ArmClient).network.VirtualWanClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_virtual_wan" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return nil
+			}
+			return err
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("Virtual WAN still exists:\n%+v", resp)
+		}
+	}
+
+	return nil
+}
+
+func testCheckAzureRMVirtualWanExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		virtualWanName := rs.Primary.Attributes["name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for Virtual WAN: %s", virtualWanName)
+		}
+
+		client := testAccProvider.Meta().(*ArmClient).network.VirtualWanClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		resp, err := client.Get(ctx, resourceGroup, virtualWanName)
+		if err != nil {
+			return fmt.Errorf("Bad: Get on virtualWanClient: %+v", err)
+		}
+
+		if resp.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: Virtual WAN %q (resource group: %q) does not exist", virtualWanName, resourceGroup)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMVirtualWan_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_wan" "test" {
+  name                = "acctestvwan%d"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  location            = "${azurerm_resource_group.test.location}"
+}
+`, rInt, location, rInt)
+}
+
+func testAccAzureRMVirtualWan_requiresImport(rInt int, location string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_virtual_wan" "import" {
+  name                = "${azurerm_virtual_wan.test.name}"
+  resource_group_name = "${azurerm_virtual_wan.test.resource_group_name}"
+  location            = "${azurerm_virtual_wan.test.location}"
+}
+`, testAccAzureRMVirtualWan_basic(rInt, location))
```

Comment:
for consistency with the other resources (so we can regex match these in the future) is it worth matching the format of the others:

```suggestion
template := testAccAzureRMVirtualWan_basic(rInt, location)
return fmt.Sprintf(`
...
`, template)
```